services.py

```diff
@@ -70,3 +72,28 @@ def update_or_create_criterion(data, criterion=None):
         criterion.deactivate()
 
     return criterion.save()
+
+
+def state_information(state):
+    links = Link.query.filter_by(state=state, active=True).all()
+
+    # https://stackoverflow.com/questions/58333162/how-to-get-top-n-results-per-group-from-a-pool-of-ids-in-sqlalchemy
+    subquery = db.session.query(
+        Score,
+        func.rank().over(
+            order_by=Score.created_at.desc(),
+            partition_by=Score.criterion_id
+        ).label('rank')
+    ).filter(Score.state == state).subquery()
+
+    scores = db.session.query(subquery).filter(subquery.c.rank <= 1).all()
+    serialized_scores = [score._asdict() for score in scores]
```

Comment:
The objects that get returned from this query are not `Score` instances but SQLAlchemy `result` instances, I think because I used the rank function in the query. I don't know if there's a way to make this return `Score` objects, but this is definitely functional for now. Let me know if you know how to fix this 😅

Reply:
I'm not too sure how to get the `Score` object! It would be nice to somehow use the serializer if we have more endpoints that follow this pattern, but it's all good for just this one :)

Reply:
We're going to have one more -- I'm going to make a ticket to figure out how to fix this (#75).